pkg/operator/controllers/subnets/subnet_nsg.go

```diff
@@ -53,18 +54,30 @@ func (r *reconcileManager) ensureSubnetNSG(ctx context.Context, s subnet.Subnet)
 	}
 	r.log.Infof("Fixing NSG from %s to %s", oldNSG, correctNSGResourceID)
 	subnetObject.NetworkSecurityGroup = &mgmtnetwork.SecurityGroup{ID: &correctNSGResourceID}
-	err = r.subnets.CreateOrUpdate(ctx, s.ResourceID, subnetObject)
-	if err != nil {
+	if err := r.subnets.CreateOrUpdate(ctx, s.ResourceID, subnetObject); err != nil {
 		return err
 	}
-
+	// Stamp the Cluster CR with a reconciliation timestamp so the e2e test picks it up.
 	return r.updateReconcileSubnetAnnotation(ctx)
 }
 
+// updateReconcileSubnetAnnotation writes the current time into the cluster annotation.
 func (r *reconcileManager) updateReconcileSubnetAnnotation(ctx context.Context) error {
 	if r.instance.Annotations == nil {
 		r.instance.Annotations = make(map[string]string)
 	}
-	r.instance.Annotations[AnnotationTimestamp] = time.Now().Format(time.RFC1123)
-	return r.client.Update(ctx, r.instance)
+	// Generate RFC1123 GMT timestamp for the e2e test’s annotation check
+	ts := time.Now().UTC().Format(time.RFC1123)
+	r.log.Infof("Annotating Cluster CR with %s=%q", AnnotationTimestamp, ts)
+
+	if _, err := time.Parse(time.RFC1123, ts); err != nil {
```

Comment:
Given that we're instantiating the time above on line 70, and the only change here is to now parse it and check for errors parsing it, I am not sure what the difference in behavior is here from what it is currently.